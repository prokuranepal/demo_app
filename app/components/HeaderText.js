import React from "react";
import { Text, StyleSheet } from "react-native";

import PropTypes from 'prop-types';

import {fontSize} from '../theme/fonts';
import {colors} from '../theme/themeColors'

const HeaderText = props => (
  <Text {...props} style={[styles.textHeading, props.style]} data-test="headerText">
    {props.children}
  </Text>
);

const styles = StyleSheet.create({
  textHeading: {
    fontSize: fontSize.header,
    color: colors.textColor1,
    fontWeight: "bold"
  }
});

HeaderText.propTypes = {
  style: PropTypes.object,
  children: PropTypes.any
}

export default HeaderText;