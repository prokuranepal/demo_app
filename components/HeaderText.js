import React from "react";
import { Text, StyleSheet } from "react-native";
import {colors} from '../ThemeColors/themeColors'
const headerText = props => (
  <Text {...props} style={[styles.textHeading, props.style]}>
    {props.children}
  </Text>
);

const styles = StyleSheet.create({
  textHeading: {
    fontSize: 26,
    color: colors.textColor1,
    fontWeight: "bold"
  }
});

export default headerText;