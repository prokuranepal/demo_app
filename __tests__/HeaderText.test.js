import React from 'react';
import {
    configure,
    shallow
} from 'enzyme';
import {Text} from 'react-native';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import HeaderText from '../src/components/HeaderText';
import {findByTestAttr} from '../src/utils/testFunction';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})

const dummy_data = {
    title:"Header",
}


describe('<HeaderText/>', () => {

    it('should render without issues', () => {
        const component = shallow(<HeaderText />);
        expect(component.length).toBe(1);
        expect(component).toMatchSnapshot();
      });

    it("Header Props test in HeaderText", () =>
      {
          const wrapper = shallow(<HeaderText>Headerrr</HeaderText>)
          expect(wrapper.find(Text)).toHaveLength(1)
          let textData = findByTestAttr(wrapper, "headerText")
          expect(textData.length).toBe(1);
          expect(textData.props().children).toEqual("Headerrr")
      
     });

})