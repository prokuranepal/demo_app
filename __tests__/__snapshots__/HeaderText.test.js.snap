// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderText/> should render without issues 1`] = `ShallowWrapper {}`;
